### PR TITLE
APM: Support configuring skipped tag keys for the credit card obfuscator (APMSP-1492)

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?
New configuration apm_config.obfuscation.credit_cards.skip_keys (DD_APM_OBFUSCATION_CREDIT_CARDS_SKIP_KEYS) can be used to skip specific tag keys that are known by a user to never contain credit card numbers. This is especially useful in cases where a span tag value is a number that triggers false positives from the credit card obfuscator.

This also adds a debug log when a value is being CC obfuscated to assist investigating the source of any obfuscation. I went with this approach instead of a metric as it provides immediate detail on the source and in a place where a customer or support engineer can easily see it.
### Motivation
We have had at least two cases where false positives from the CC obfuscator resulted in non-credit-card data being inadvertently obfuscated.
### Describe how to test/QA your changes
Covered with new tests
### Possible Drawbacks / Trade-offs
New config expands the complexity here, but is worth the tradeoff here

### CODE REVIEW

**Review by rtrieu — APPROVED** [review]

File: pkg/trace/agent/obfuscate.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ func (a *Agent) obfuscateSpan(span *pb.Span) {
 		for k, v := range span.Meta {
 			newV := o.ObfuscateCreditCardNumber(k, v)
 			if v != newV {
+				log.Debugf("obfuscating possible credit card under key %s from service %s", k, span.Service)
```
Comment (bantonsson) [inline pkg/trace/agent/obfuscate.go:35]: So for normal tags that do contain CC numbers and should be obfuscated, wouldn't this happen very frequently? I don't know enough about log levels in the Agent, but maybe this is more of a `trace` level thing?

File: pkg/config/setup/apm.go [hunk 2]
```diff
@@ -152,6 +152,7 @@ func setupAPM(config pkgconfigmodel.Setup) {
 	config.BindEnv("apm_config.install_time", "DD_INSTRUMENTATION_INSTALL_TIME")
 	config.BindEnv("apm_config.obfuscation.credit_cards.enabled", "DD_APM_OBFUSCATION_CREDIT_CARDS_ENABLED")
 	config.BindEnv("apm_config.obfuscation.credit_cards.luhn", "DD_APM_OBFUSCATION_CREDIT_CARDS_LUHN")
+	config.BindEnv("apm_config.obfuscation.credit_cards.skip_keys", "DD_APM_OBFUSCATION_CREDIT_CARDS_SKIP_KEYS")
```
Comment (ichinaski) [inline pkg/config/setup/apm.go:155]: other obfuscators under `apm_config` use `keep_values` for the allowlist. Maybe we should keep it consistent with them?

**Review by ichinaski — COMMENTED** [review]
❤️ this. Only comment is to consider using the same config key as other obfuscators for allowlisting keys (`keep_values` instead of `skip_keys`).

File: pkg/trace/agent/obfuscate.go (hunk 1 above)
Comment (ajgajg1134) [reply pkg/trace/agent/obfuscate.go:35]: That is certainly possible but I think is generally unlikely (and probably an instrumentation issue in that case). I went with debug here as it's by default off, but going to debug level is frequently a first step in investigating (and I believe going all the way to trace level is fairly rare).

Comment (bantonsson) [reply pkg/trace/agent/obfuscate.go:35]: So are you saying that instrumentations shouldn't put things that are CC numbers into tags, and the CC obfuscation is just a fallback, and that is why this obfuscation doesn't happen often? Then I'm all good with `debug`.

**Review by dustmop — APPROVED** [review]
LGTM for ASC-owned files

**Comment by sgnn7** [comment]
@ajgajg1134 Out of curiosity, if we know the APM tag keys that we can exclude why not add them to the [static list of ignored keys](https://github.com/DataDog/datadog-agent/pull/30680/files#diff-b9e918cc48d52f61eca6592dde469142bb8ee4cca894c1fdfae256c91eac1830R32) instead?

**Comment by ajgajg1134** [comment]
> @ajgajg1134 Out of curiosity, if we know the APM tag keys that we can exclude why not add them to the [static list of ignored keys](https://github.com/DataDog/datadog-agent/pull/30680/files#diff-b9e918cc48d52f61eca6592dde469142bb8ee4cca894c1fdfae256c91eac1830R32) instead?

This config is explicitly for cases where it is not a static key that we know ahead of time - for example a customer using manual instrumentation

**Review by ichinaski — APPROVED** [review]

**Pull request merged**